cli/provider/cmd.go

```diff
@@ -206,23 +227,35 @@ func (c *CmdConfigurator) AddFlags(cmd *cobra.Command) error {
 		}
 
 	case "record", "test", "rerecord":
-		cmd.Flags().StringP("path", "p", ".", "Path to local directory where generated testcases/mocks are stored")
-		cmd.Flags().Uint32("proxy-port", c.cfg.ProxyPort, "Port used by the Keploy proxy server to intercept the outgoing dependency calls")
-		cmd.Flags().Uint32("dns-port", c.cfg.DNSPort, "Port used by the Keploy DNS server to intercept the DNS queries")
-		cmd.Flags().StringP("command", "c", c.cfg.Command, "Command to start the user application")
-
-		cmd.Flags().String("cmd-type", c.cfg.CommandType, "Type of command to start the user application (native/docker/docker-compose)")
-		cmd.Flags().Uint64P("build-delay", "b", c.cfg.BuildDelay, "User provided time to wait docker container build")
-		cmd.Flags().String("container-name", c.cfg.ContainerName, "Name of the application's docker container")
-		cmd.Flags().StringP("network-name", "n", c.cfg.NetworkName, "Name of the application's docker network")
-		cmd.Flags().UintSlice("pass-through-ports", config.GetByPassPorts(c.cfg), "Ports to bypass the proxy server and ignore the traffic")
-		cmd.Flags().Uint64P("app-id", "a", c.cfg.AppID, "A unique name for the user's application")
-		cmd.Flags().String("app-name", c.cfg.AppName, "Name of the user's application")
-		cmd.Flags().Bool("generate-github-actions", c.cfg.GenerateGithubActions, "Generate Github Actions workflow file")
-		cmd.Flags().Bool("in-ci", c.cfg.InCi, "is CI Running or not")
-		//add rest of the uncommon flags for record, test, rerecord commands
-		c.AddUncommonFlags(cmd)
-
+		if cmd.Parent() != nil && cmd.Parent().Name() == "contract" {
+			cmd.Flags().StringSliceP("services", "s", c.cfg.Contract.Services, "Specify the services for which to generate contracts")
+			cmd.Flags().StringP("path", "p", c.cfg.Contract.Path, "Specify the path to generate contracts")
+			if c.cfg.Contract.Path == "" {
+				c.cfg.Contract.Path = "./"
+			}
+			cmd.Flags().Bool("download", c.cfg.Contract.Download, "Specify whether to download contracts or not")
+			cmd.Flags().Bool("generate", c.cfg.Contract.Generate, "Specify")
+			cmd.Flags().String("driven", c.cfg.Contract.Driven, "Specify the driven flag to validate contracts")
```

Comment:
driven flag is required.

Reply:
Or maybe we can just take consumer as default so there is no need to set it required